pkg/storage/cacher.go

```diff
@@ -464,7 +467,15 @@ func (c *cacheWatcher) stop() {
 }
 
 func (c *cacheWatcher) add(event watchCacheEvent) {
-	c.input <- event
+	select {
+	case c.input <- event:
+	case <-time.After(time.Second):
```

Comment:
Maybe we should increase timeout here?
If we set it to e.g. 5 seconds, hitting this timeout should be very rare, I think...